osc/fetch.py

```diff
@@ -221,12 +221,22 @@ def run(self, buildinfo):
                 cached += 1
                 if not i.name.startswith('container:') and i.pacsuffix != 'rpm':
                     continue
+
+                hdrmd5_is_valid = True
                 if i.hdrmd5:
                     if i.name.startswith('container:'):
                         hdrmd5 = dgst(i.fullfilename)
+                        if not hdrmd5 or hdrmd5 != i.hdrmd5:
+                            hdrmd5_is_valid = False
                     else:
                         hdrmd5 = packagequery.PackageQuery.queryhdrmd5(i.fullfilename)
-                    if not hdrmd5 or hdrmd5 != i.hdrmd5:
+                        if not hdrmd5 or hdrmd5 != i.hdrmd5:
+                            # FIXME: We're currently ignoring the hdrmd5 verification of rpm packages.
+                            # look for "HDRMD5 MISMATCH EXPLANATION" below
+                            print("Warning: Ignoring a hdrmd5 mismatch for %s: %s (actual) != %s (expected)" % (i.fullfilename, hdrmd5, i.hdrmd5))
+                            hdrmd5_is_valid = True
```

Comment:
this should be really only done via a switch. Or one looks endless for issues which are not reproducible locally due to broken noarch rpms (which are in fact not noarch)

Reply:
Agreed. `disable_hdrmd5_check` is going to be part of the changes I'm about to force-push.